server/interpreter.js

```diff
@@ -3303,18 +3064,17 @@ Interpreter.prototype['stepUnaryExpression'] = function() {
     }
     value = this.deleteProperty(obj, name);
     if (!value) {
-      this.throwException(this.TYPE_ERROR, 'Cannot delete property \'' +
-                          name + '\' of \'' + obj + '\'');
-      return;
+      this.throwException(this.TYPE_ERROR, "Cannot delete property '" +
+                          name + "' of '" + obj + "'");
     }
-  } else if (node['operator'] == 'typeof') {
-    value = value.type;
-  } else if (node['operator'] == 'void') {
+  } else if (node['operator'] === 'typeof') {
+    value = value.node ? 'function' : typeof value;
```

Comment:
This should identify user functions (and therefore polyfills), but will it correctly identify native functions?  Perhaps you should check `.isFunction`?